src/paint.rs

```diff
@@ -140,13 +142,100 @@ impl<'a> Painter<'a> {
             Self::get_diff_style_sections(&self.minus_lines, &self.plus_lines, self.config);
 
         if self.config.side_by_side {
+            // Only set `wrap_lines` to true if wrapping is wanted and lines which are
+            // too long are found.
+            // If so, remember the calculated line width and which of the lines are too
+            // long for later re-use.
```

Comment:
I believe that the two calls below to `wrap::minus_plus_too_long` will always return the same values. That is, the following assertions hold (after adding a `PartialEq` implementation to `PlusMinus`):

```rust
assert!(wrap_syntax == wrap_diff);
assert!(syntax_style_lines_too_long.minus == diff_style_lines_too_long.minus);
assert!(syntax_style_lines_too_long.plus == diff_style_lines_too_long.plus);
```

The reasons these are the same is that the `*_style_sections` data structures are arrays of `(style, substring)` tuples, and the concatenation of the `substring`s is the same in both cases. I.e. `*_syntax_style_sections` and `*_diff_style_sections` represent the same line, broken up in different ways.

Since these are (I believe) guaranteed to be the same at runtime, I think it might make the code easier to understand if we just have one variable name for each quantity.